### PR TITLE
feat: reduce size CSS output file

### PR DESCRIPTION
**Pull request by ghassanmas**
This is ready again for review, see below for testing detials and result

~~[Update] 18.01.22 As of testing with olive MFE learning, the CSS output was missing a lot of stuff. Converted to draft again, need to double check the correct settings~~

 The added plugin, reduce the CSS file by removing unused CSS
 The plugin is kinda recommended by bootstrap[^1], reducing
 unused CSS is generally recommended as well enchance web
 perforamce[^2]

 [^1]: https://getbootstrap.com/docs/5.2/customize/optimize/#unused-css
 [^2]: https://web.dev/css-web-vitals/#critical-css

 Related issue: openedx/frontend-wg/issues/138

This is still work on progress, I need to do further testing, to ensure this change doesn't break styling or any other thing

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ghassanmas!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by itsjeyd** [comment]
@ghassanmas Thanks for your contribution! Please let us know when it's ready for review.

**Comment by itsjeyd** [comment]
Hey @jmbowman, this is ready for engineering review by the FED-BOM team now.

**Review by abdullahwaheed — CHANGES_REQUESTED** [review]

**Comment by ghassanmas** [comment]
[Update]: 

Testing: I have tested this version on frontend-app-learning, by installing this version of frontend-build: see this PR https://github.com/openedx/frontend-app-learning/pull/1047 

The effect of this is that the css way more than expected. I don't rememeber how I test it when I opened the draft, but in any case this cannon be merged as it is.  I am going convert it to draft again, and when time allow I will dig into the correct settings. s

**Comment by ghassanmas** [comment]
[Update] I think I made it to work with learning MFE olive, CSS file size was decreased 70% (few hundred kilo bytes). I had to completely set it in a different way, please check the expected diff at https://github.com/ghassanmas/frontend-build/pull/1 . If that can of change acceptable I would tidy up and update this PR accordingly,

**Comment by ghassanmas** [comment]
# The test on MFE Learning

## Before This change 

**CSS File size**: 132KB GIZ /883K/Original

image

Live link: https://apps.olive.zaat.dev/learning/course/course-v1:zaatdev+101+1/home


## After This change 

image

**CSS File size**:  27KB GIZPPED/  278K Orginal 

Live Link: https://learning.olive.zaat.dev/course/course-v1:zaatdev+101+1/home


## Concolusion 

 Increaing [lighthouse](https://web.dev/measure/) performance by 20%

**Comment by abdullahwaheed** [comment]
Its improving the build size but build time is affecting greatly with these changes. I have tested it on multiple MFEs and following are the results:
**frontend-app-account**:
    without css optimization -> build time: 65974 ms
    with optimization -> build time: 150820 ms

**frontend-app-learner-portal-enterprise**:
    without css optimization -> build time: 73184 ms
    with optimization -> build time: 268651 ms

**frontend-app-learning**:
    without css optimization -> build time: 121364 ms
    with optimization -> build time: 239722 ms


@arbrandes what are your thoughts on this?

**Comment by ghassanmas** [comment]
Yes I expected that would occuar, but for me I think the priority comes for end user which is leaner. 

For example at 2U/edx you do one build for milllions of users, compare those few seconds with a fraction of second of for million of users.

**Comment by arbrandes** [comment]
I have to be honest: I don't much like CSS removers.  There's too much risk for too little reward.  This article outlines some of the dangers:

https://css-tricks.com/how-do-you-remove-unused-css-from-a-site/#aa-part-of-a-build-process

The added build time is also bad.  We're trying to reduce it for many reasons, and this will make it much worse.  I don't think saving 100KB of downloads is worth it.

What I'd be more interested in is ways to avoid having to insert framework CSS when it's not necessary in the first place, possibly by picking more modular frameworks.

**Comment by itsjeyd** [comment]
@ghassanmas A friendly reminder to follow up on @arbrandes's comment above.

**Comment by ghassanmas** [comment]
@itsjeyd we have discussed this in the last FWG, https://openedx.atlassian.net/wiki/spaces/COMM/pages/3645964289/2023-01-26+Frontend+Working+Group+Meeting+Minutes. And I think @arbrandes want to check with folks opinion before merging it.

**Comment by itsjeyd** [comment]
Thanks for the info @ghassanmas!

As of today, what do you think will be the next steps for this PR? Is it still waiting for results from this:

> I think @arbrandes want to check with folks opinion before merging it.

... Or are you considering to follow @regisb's suggestion and reimplement the changes as a plugin (in which case it might make sense to close this PR)?

**Comment by ghassanmas** [comment]
Thinking about this, I came with two scnario: 
1. implement a featuer flag to enable/disable adding this plugin 
   - it can be by default off which should not affect tutor/tutor-mfe
1.  Doing a tutor plugin for that is indeed possiblem but it would make it a bit of hassle to maintain ** a plugin and a fork of front-build**

Overall the bigger picture of this I think we ideally tutor-mfe can provide an image which is built once for all other providers, and it could have feature flag of which when its turned on **it would make build faster but not optimize the build size** [^1]. The scinario I have in mind is that developer would keep rebuilding the image until they see the changes they want, and then once it looks good we can advise them to rebuild while turnning all optimization flag

 [^1]: For that to we might not only skip this plugin but also other plugin that make the build slow, see overhangio/tutor-mfe/issues/88 for example.

**Comment by regisb** [comment]
> Doing a tutor plugin for that is indeed possible but it would make it a bit of hassle to maintain a plugin and a fork of front-build

Could we avoid forking frontend-build? If I understand correctly all we have to do is to modify webpack.config.js, right? So we could ship our own version of that file, just like we do in development.

That being said, my initial knee-jerk reaction is the same as Adolfo: CSS removal looks tricky and I'd be afraid to break too many things.

**Comment by regisb** [comment]
I realize now that my [previous comment](https://github.com/openedx/frontend-build/pull/266#issuecomment-1416056973) was stupid. I initially thought that this issue was part of the tutor-mfe plugin, and not frontend-build :sweat: Thus my comment about making this change part of tutor-mfe does not really make sense anymore...

IMHO the compressed size decrease is pretty serious, and it's worth considering. But I'm wondering if the proposed fix is not the tree that hides the forest. I mean look at the numbers: are we seriously generating a CSS file that weighs 883kB? What for? Clearly, this pull request shows that there is a ton of crap in this file (69% to be precise). The word "bloated" comes to mind. Why is there so much stuff in this file in the first place? I'm afraid that if we use a plugin to purge this file we'll just blindingly adding more and more stuff, and in a couple months/years down the road we'll be facing the same problem again.

**Comment by jmbowman** [comment]
Building on the last comment - can we use the purgecss output to identify the stuff being ripped out, and try to avoid including it in the first place?  Maybe via appropriate Bootstrap or Paragon configuration options, etc.

**Comment by ghassanmas** [comment]
Inclose is the file of git diff between the original css compared to the newly generated.

The generation of such `git diff`
- run the build with my patch `npm run build`
  - call the dist file something else to ensure its not earsed can coexists with `mv dist dist-with`
- run the build without my patch `npm run build` _now we have dist(the typical bundle) and dist-with_ 
so running the following command would generate the uploaded file 
` git diff --no-index --clolor dist/app.4fce923d8a1007f74bd7.css dist-with/app.2abb1105ce90dd2f22da.css > purge_diff.txt`  [^1]


Here is a sippest of what example of css that are removed 
```css
- */@font-face{font-family:FontAwesome;font-style:normal;font-weight:400;src:url(/8b43027f47b20503057dfbbaa9401fef.eot);src:url(/8b43027f47b20503057dfbbaa9401fef.eot?#iefix&v=4.7.0) format("embedded-opentype"),url(/20fd1704ea223900efa9fd4e869efb08.woff2) format("woff2"),url(/f691f37e57f04c152e2315ab7dbad881.woff) format("woff"),url(/1e59d2330b4c6deb84b340635ed36249.ttf) format("truetype"),url(/c1e38fd9e0e74ba58f7a2b77ef29fdd3.svg#fontawesomeregular) format("svg")}.fa{text-rendering:auto;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;display:inline-block;font:normal normal normal 14px/1 FontAwesome;font-size:inherit}.fa-lg{font-size:1.3333333333em;line-height:.75em;vertical-align:-15%}.fa-2x{font-size:2em}.fa-3x{font-size:3em}.fa-4x{font-size:4em}.fa-5x{font-size:5em}.fa-fw{text-align:center;width:1.2857142857em}.fa-ul{list-style-type:none}[dir=ltr] .fa-ul{margin-left:2.1428571429em;padding-left:0}[dir=rtl] .fa-ul{margin-right:2.1428571429em;padding-right:0}.fa-ul>li{position:relative}.fa-li{position:absolute;text-align:center;top:.1428571429em;width:2.1428571429em}[dir=ltr] .fa-li{left:-2.1428571429em}[dir=rtl] .fa-li{right:-2.1428571429em}[dir=ltr] .fa-li.fa-lg{left:-1.8571428571em}[dir=rtl] .fa-li.fa-lg{right:-1.8571428571em}.fa-border{border:.08em solid #eee;border-radius:.1em;padding:.2em .25em .15em}[dir=ltr] .fa-pull-left{float:left}[dir=ltr] .fa-pull-right,[dir=rtl] .fa-pull-left{float:right}[dir=rtl] .fa-pull-right{float:left}[dir=ltr] .fa.fa-pull-left{margin-right:.3em}[dir=ltr] .fa.fa-pull-right,[dir=rtl] .fa.fa-pull-left{margin-left:.3em}[dir=rtl] .fa.fa-pull-right{margin-right:.3em}[dir=ltr] .pull-right{float:right}[dir=ltr] .pull-left,[dir=rtl] .pull-right{float:left}[dir=rtl] .pull-left{float:right}[dir=ltr] .fa.pull-left{margin-right:.3em}[dir=ltr] .fa.pull-right,[dir=rtl] .fa.pull-left{margin-left:.3em}[dir=rtl] .fa.pull-right{margin-right:.3em}.fa-spin{animation:fa-spin 2s linear infinite}.fa-pulse{animation:fa-spin 1s steps(8) infinite}@keyframes fa-spin{0%{transform:rotate(0deg)}to{transform:rotate(359deg)}}.fa-rotate-90{-ms-filter:"progid:DXImageTransform.Microsoft.BasicImage(rotation=1)"}[dir=ltr] .fa-rotate-90{transform:rotate(90deg)}[dir=rtl] .fa-rotate-90{transform:rotate(-90deg)}.fa-rotate-180{-ms-filter:"progid:DXImageTransform.Microsoft.BasicImage(rotation=2)"}[dir=ltr] .fa-rotate-180{transform:rotate(180deg)}[dir=rtl] .fa-rotate-180{transform:rotate(-180deg)}.fa-rotate-270{-ms-filter:"progid:DXImageTransform.Microsoft.BasicImage(rotation=3)"}[dir=ltr] .fa-rotate-270{transform:rotate(270deg)}[dir=rtl] .fa-rotate-270{transform:rotate(-270deg)}.fa-flip-horizontal{-ms-filter:"progid:DXImageTransform.Microsoft.BasicImage(rotation=0, mirror=1)";transform:scaleX(-1)}.fa-flip-vertical{-ms-filter:"progid:DXImageTransform.Microsoft.BasicImage(rotation=2, mirror=1)";transform:scaleY(-1)}:root .fa-flip-horizontal,:root .fa-flip-vertical,:root .fa-rotate-180,:root .fa-rotate-270,:root .fa-rotate-90{filter:none}.fa-stack{display:inline-block;height:2em;line-height:2em;position:relative;vertical-align:middle;width:2em}.fa-stack-1x,.fa-stack-2x{position:absolute;text-align:center;width:100%}[dir=ltr] .fa-stack-1x,[dir=ltr] .fa-stack-2x{left:0}[dir=rtl] .fa-stack-1x,[dir=rtl] .fa-stack-2x{right:0}.fa-stack-1x{line-height:inherit}.fa-stack-2x{font-size:2em}.fa-inverse{color:#fff}.fa-glass:before{content:"�"}.fa-music:before{content:"�"}.fa-search:before{content:"�"}.fa-envelope-o:before{content:"�"}.fa-heart:before{content:""}.fa-star:before{content:""}.fa-star-o:before{content:""}.fa-user:before{content:""}.fa-film:before{content:""}.fa-th-large:before{content:""}.fa-th:before{content:""}.fa-th-list:before{content:""}.fa-check:before{content:""}.fa-close:before,.fa-remove:before,.fa-times:before{content:""}.fa-search-plus:before{content:""}.fa-search-minus:before{content:""}.fa-power-off:before{content:""}.fa-signal:before{content:""}.fa-cog:before,.fa-gear:before{content:""}.fa-trash-o:before{content:""}.fa-home:before{content:""}.fa-file-o:before{content:""}.fa-clock-o:before{content:""}.fa-road:before{content:""}.fa-download:before{content:""}.fa-arrow-circle-o-down:before{content:""}.fa-arrow-circle-o-up:before{content:""}.fa-inbox:before{content:""}.fa-play-circle-o:before{content:""}.fa-repeat:before,.fa-rotate-right:before{content:""}.fa-refresh:before{content:""}.fa-list-alt:before{content:""}.fa-lock:before{content:""}.fa-flag:before{content:""}.fa-headphones:before{content:""}.fa-volume-off:before{content:""}.fa-volume-down:before{content:""}.fa-volume-up:before{content:""}.fa-qrcode:before{content:""}.fa-barcode:before{content:""}.fa-tag:before{content:""}.fa-tags:before{content:""}.fa-book:before{content:""}.fa-bookmark:before{content:""}.fa-print:before{content:""}.fa-camera:before{content:""}.fa-font:before{content:""}.fa-bold:before{content:""}.fa-italic:before{content:""}.fa-t:""}.fa-print:before{content:""}.fa-camera:before{content:""}.fa-font:before{content:""}.fa-bold:before{content:""}.fa-italic:before{content:""}.fa-
text-height:before{content:""}.fa-text-width:before{content:""}.fa-align-left:before{content:""}.fa-align-center:before{content:""}.fa-align-right:before{content:""}.fa-align-justify:before{content:""}.fa-list:before{content:""}.fa-dedent:before,.fa-outdent:before{content:""}.fa-indent:before{content:
""}.fa-video-camera:before{content:""}.fa-image:before,.fa-photo:before,.fa-picture-o:before{content:""}.fa-pencil:before{content:""}.fa-map-marker:bef
ore{content:""}.fa-adjust:before{content:""}.fa-tint:before{content:""}.fa-edit:before,.fa-pencil-square-o:before{content:""}.fa-share-square-o:before{
content:""}.fa-check-square-o:before{content:""}.fa-arrows:before{content:""}.fa-step-backward:before{content:""}.fa-fast-backward:before{content:""}.
fa-backward:before{content:""}.fa-play:before{content:""}.fa-pause:before{content:""}.fa-stop:before{content:""}.fa-forward:before{content:""}.fa-fast
-forward:before{content:""}.fa-step-forward:before{content:""}.fa-eject:before{content:""}.fa-chevron-left:before{content:""}.fa-chevron-right:before{c
ontent:""}.fa-plus-circle:before{content:""}.fa-minus-circle:before{content:""}.fa-times-circle:before{content:""}.fa-check-circle:before{content:""}.
fa-question-circle:before{content:""}.fa-info-circle:before{content:""}.fa-crosshairs:before{content:""}.fa-times-circle-o:before{content:""}.fa-check-
circle-o:before{content:""}.fa-ban:before{content:""}.fa-arrow-left:before{content:""}.fa-arrow-right:before{content:""}.fa-arrow-up:before{content:""
}.fa-arrow-down:before{content:""}.fa-mail-forward:before,.fa-share:before{content:""}.fa-expand:before{content:""}.fa-compress:before{content:""}.fa-p
lus:before{content:""}.fa-minus:before{content:""}.fa-asterisk:before{content:""}.fa-exclamation-circle:before{content:""}.fa-gift:before{content:""}.
fa-leaf:before{content:""}.fa-fire:before{content:""}.fa-eye:before{content:""}.fa-eye-slash:before{content:""}.fa-exclamation-triangle:before,.fa-warn
ing:before{content:""}.fa-plane:before{content:""}.fa-calendar:before{content:""}.fa-random:before{content:""}.fa-comment:before{content:""}.fa-magnet
:before{content:""}.fa-chevron-up:before{content:""}.fa-chevron-down:before{content:""}.fa-retweet:before{content:""}.fa-shopping-cart:before{content:"
"}.fa-folder:before{content:""}.fa-folder-open:before{content:""}.fa-arrows-v:before{content:""}.fa-arrows-h:before{content:""}.fa-bar-chart-o:before,
.fa-bar-chart:before{content:""}.fa-twitter-square:before{content:""}.fa-facebook-square:before{content:""}.fa-camera-retro:before{content:""}.fa-key:b
efore{content:""}.fa-cogs:before,.fa-gears:before{content:""}.fa-comments:before{content:""}.fa-thumbs-o-up:before{content:""}.fa-thumbs-o-down:before{
content:""}.fa-star-half:before{content:""}.fa-heart-o:before{content:""}.fa-sign-out:before{content:""}.fa-linkedin-square:before{content:""}.fa-thum
b-tack:before{content:""}.fa-external-link:before{content:""}.fa-sign-in:before{content:""}.fa-trophy:before{content:""}.fa-github-square:before{conten
t:""}.fa-upload:before{content:""}.fa-lemon-o:before{content:""}.fa-phone:before{content:""}.fa-square-o:before{content:""}.fa-bookmark-o:before{conte
nt:""}.fa-phone-square:before{content:""}.fa-twitter:before{content:""}.fa-facebook-f:before,.fa-facebook:before{content:""}.fa-github:before{content:"
"}.fa-unlock:before{content:""}.fa-credit-card:before{content:""}.fa-feed:before,.fa-rss:before{content:""}.fa-hdd-o:before{content:""}.fa-bullhorn:be
fore{content:""}.fa-bell:before{content:""}.fa-certificate:before{content:""}.fa-hand-o-right:before{content:""}.fa-hand-o-left:before{content:""}.fa-
hand-o-up:before{content:""}.fa-hand-o-down:before{content:""}.fa-arrow-circle-left:before{content:""}.fa-arrow-circle-right:before{content:""}.fa-arro
w-circle-up:before{content:""}.fa-arrow-circle-down:before{content:""}.fa-globe:before{content:""}.fa-wrench:before{content:""}.fa-tasks:before{content
:""}.fa-filter:before{content:""}.fa-briefcase:before{content:""}.fa-arrows-alt:before{content:""}.fa-group:before,.fa-users:before{content:""}.fa-cha
in:before,.fa-link:before{content:""}.fa-cloud:before{content:""}.fa-flask:before{content:""}.fa-cut:before,.fa-scissors:before{content:""}.fa-copy:bef
ore,.fa-files-o:before{content:""}.fa-paperclip:before{content:""}.fa-floppy-o:before,.fa-save:before{content:""}.fa-square:before{content:""}.fa-bars:
before,.fa-navicon:before,.fa-reorder:before{content:""}.fa-list-ul:before{content:""}.fa-list-ol:before{content:""}.fa-strikethrough:before{content:""
}.fa-underline:before{content:""}.fa-table:before{content:""}.fa-magic:before{content:""}.fa-truck:before{content:""}.fa-pinterest:before{content:""}.
```
I speculate the reason is that there import of classes/modules that are not used, 'fa-github:before:` is that calss used in frontend-app-account?,

- I think the typical way to resolve this kind of issue is to do tree-shaking, but I am not sure tree shaking also support css/scss file, because usually tree-shaking in JS modules context. 

- Other suggestion, _not sure its 100%_ is to ensure we are only importing classes that being used, so webpack doesn't look for all classes/files. 

[^1]: [purge_diff.txt](https://github.com/openedx/frontend-build/files/10724097/pruge_diff.txt)

**Comment by itsjeyd** [comment]
Hey @ghassanmas, just checking in to see if this PR is still needed?

**Comment by ghassanmas** [comment]
@itsjeyd I think the ball is in @arbrandes field

**Comment by itsjeyd** [comment]
Thanks @ghassanmas, noted.

Hey @arbrandes, when you get a minute could you let us know when you're planning to have another look at this PR?

**Comment by arbrandes** [comment]
@ghassanmas, I think the idea is sound.  Whether or not we succeed in not including things we don't need before the build step, there will likely always be some things that end up being build but unused.

However, due to the fact this is not without cost, we should make it optional via build-time variable (then later support this variable in [tutor-mfe](https://github.com/overhangio/tutor-mfe)).  Do you mind adding that in, Ghassan?

File: config/webpack.prod.config.js
Comment (arbrandes) [inline config/webpack.prod.config.js:20]: The codebase hasn't been updated in years, apparently.  Are there no more active alternatives?

**Review by arbrandes — CHANGES_REQUESTED** [review]
As mentioned above, the idea is good, but we should develop it further.

File: config/webpack.prod.config.js
Comment (ghassanmas) [reply config/webpack.prod.config.js:20]: Hum... This installs `@fullhuman/postcss-purgecss` version > 5.x which was publi[shed 6 months ago in npm](https://www.npmjs.com/package/@fullhuman/postcss-purgecss) hence the last git commit https://github.com/FullHuman/purgecss/commits/main/packages/postcss-purgecss repo is quite acitve. 
Probably you were looking to something else, as it's easy to find pkgs with similar name

**Comment by ghassanmas** [comment]
This is again ready for review: 
- It's now optional to use prugecss, readme.rst has been updated _by default it's not used_
- It include `ts` and `tsx` file checks

**Comment by ghassanmas** [comment]
@arbrandes following your suggestion I made it optional, and as for tutor-mfe I geuss it worth mentioning this option, and I would suggest that the default tutor-mfe image to use since it would be build once and used many times.

**Comment by itsjeyd** [comment]
label: core contributor

**Review by abdullahwaheed — APPROVED** [review]
LGTM

File: README.rst
Comment (arbrandes) [inline README.rst:189]:
```suggestion
On the other hand, enabling PurgeCSS will increase browser performance for the end user by as much as 20% (as measured by `lighthouse`_).  Operators are encouraged to do so for production deployments.

For more information about optimizing MFEs, refer to the `issue #138`_ in the wg-frontend repository.
```

File: README.rst
Comment (arbrandes) [inline README.rst:192]:
```suggestion
.. issue #138: https://github.com/openedx/wg-frontend/issues/138
```

File: README.rst
Comment (arbrandes) [inline README.rst:188]:
```suggestion
However, note that doing this will increase build time by 30%. It's thus not recommended to use this option during development.
```

File: config/webpack.prod.config.js
Comment (arbrandes) [inline config/webpack.prod.config.js:123]: I think it would improve readability (and future extensibility) to use the `extraPlugins` pattern already used in this file.  For example, you could do something like:

```javascript
const extraPostCssPlugins = [];
if (process.env.USE_PURGESS !== 'false') {
  extraPostCssPlugins.push(purgecss({
    content: ['./**/*.html', './**/*.js', './**/*.jsx', './**/*.ts', './**/*.tsx'],
  }));
}
```

Then:

```javascript
                plugins: [
                  PostCssAutoprefixerPlugin(),
                  PostCssRTLCSS(),
                  CssNano(),
                  PostCssCustomMediaCSS(),
                  ...extraPostCssPlugins,
                ],
```

File: config/webpack.prod.config.js
Comment (arbrandes) [inline config/webpack.prod.config.js:202]:
```suggestion
```

File: package.json
Comment (arbrandes) [inline package.json:82]: `@` comes first alphabetically, so we should put it at the top.

File: package.json
Comment (arbrandes) [inline package.json:91]: Needs alphabetical ordering here as well.

**Review by arbrandes — CHANGES_REQUESTED** [review]
Looks good in principle, but I think the code could still benefit from a couple of changes.

Thanks for bearing with me, @ghassanmas.

File: config/webpack.prod.config.js
Comment (ghassanmas) [reply config/webpack.prod.config.js:123]: Make sense but just to double confirm we are on the same page, would you prefer to by deafult this would be enabled or disabled?

File: package.json
Comment (ghassanmas) [reply package.json:91]: Good catch, I think I should remove it alotgather, it was used in the first draft then I switch to @fullhuman/..etc

**Comment by ghassanmas** [comment]
@arbrandes no worries, I did your suggestion and have rebased to single commit. Have tested again: 

- `USE_PURGECSS=true npm run build` Would enable it reduces app.*.css file by size and increase build time
- `npm run build` nothing changes 

Testing while applying the following patch in account MFE: 

```git
-    "@edx/frontend-build": "9.1.1",
+    "@edx/frontend-build": "git+http://github.com/ghassanmas/frontend-build.git#6680a856779fd8a204ef290af3d1f658e4e3f3e6",
     "@edx/reactifex": "^1.0.3",

```

**Comment by itsjeyd** [comment]
@ghassanmas Looks like this is getting closer to being merged :)

Could you please resolve merge conflicts and ping @arbrandes for another round of review when done?

**Comment by itsjeyd** [comment]
Thanks @ghassanmas!

@arbrandes Back to you :)

**Review by arbrandes — APPROVED** [review]
Looks great!  As usual, thank you for you patience, @ghassanmas, and congrats on pushing this to completion.

**Comment by openedx-webhooks** [comment]
@ghassanmas 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-build&entry.867055334=https://github.com/openedx/frontend-build/pull/266&entry.1484655318=https://github.com/ghassanmas&entry.752974735=2022-10-27+09:16&entry.1917517419=2023-04-04+16:07&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
